smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServiceGenerator.java

```diff
@@ -238,6 +238,9 @@ private void generateClientDefaults() {
             writer.writeDocs("The runtime environment");
             writer.write("runtime?: string;\n");
 
+            writer.writeDocs("Disable setting host prefix");
```

Comment:
This might need a little more information. Maybe add something like "Disable dyanamically changing the endpoint of the client based on the hostPrefix trait of an operation."